<a href="https://colab.research.google.com/github/saimahesh16/adv-pprog/blob/master/assignment_28_05_20.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Read the input as two limits and Print the output as Palindrome count between the two limits
Example #1; Input : 1 10 Output : 9
 Example #2: Input : 1 100 Output : 18


In [4]:
def isPalindrome(n: int) -> bool: 
  
    # Find reverse of n 
    rev = 0
    i = n 
    while i > 0: 
        rev = rev * 10 + i % 10
        i //= 10
  
    # If n and rev are same,  
    # then n is palindrome 
    return (n == rev) 
  
# prints palindrome between min and max 
def countPal(minn: int, maxx: int) -> None: 
    for i in range(minn, maxx + 1): 
        if isPalindrome(i): 
            print(i, end = " ") 
  
# Driver Code 
if __name__ == "__main__": 
    countPal(100, 2000) 

101 111 121 131 141 151 161 171 181 191 202 212 222 232 242 252 262 272 282 292 303 313 323 333 343 353 363 373 383 393 404 414 424 434 444 454 464 474 484 494 505 515 525 535 545 555 565 575 585 595 606 616 626 636 646 656 666 676 686 696 707 717 727 737 747 757 767 777 787 797 808 818 828 838 848 858 868 878 888 898 909 919 929 939 949 959 969 979 989 999 1001 1111 1221 1331 1441 1551 1661 1771 1881 1991 

Write a program to prompt for a score between 0.0 and 1.0. If the score is out of range, print an error. If the score is between 0.0 and 1.0, print a grade using the following table:
Score Grade
>= 0.9 A
>= 0.8 B
>= 0.7 C
>= 0.6 D
< 0.6 F
If the user enters a value out of range, print a suitable error message and exit. For the test, enter a score of 0.85


In [6]:
score = input("Enter Score: ")
s =  float(score)
x = 'Error'
if s >= 0.9:
	x = 'A'
elif s >=0.8:
	x='B'
elif s >=0.7:
	x='C'
elif s >= 0.6:
	x='D'
elif s < .6:
	x ='F'
else:
	x ="Out of Range"
print (x)

Enter Score: 0.85
B


Write a Python program to find the median of three values.
Expected Output:
Input first number: 26                                            
Input second number: 15                                                
Input third number: 29                                                  
The median is 26


In [1]:
a = int(input("Input first number: "))
b = int(input("Input second number: "))
c = int(input("Input third number: "))
if a > b:
    if a < c:
        median = a
    elif b > c:
        median = b
    else:
        median = c
else:
    if a > c:
        median = a
    elif b < c:
        median = b
    else:
        median = c

print("The median is", median)

Input first number: 6
Input second number: 2
Input third number: 5
The median is 5


Write a Python program which iterates the integers from 1 to 50. For multiples of three print "Fizz" instead of the number and for the multiples of five print "Buzz". For numbers which are multiples of both three and five print "FizzBuzz".
Sample Output :
fizzbuzz
1
2
fizz
4
buzz


In [2]:
for fizzbuzz in range(0,51):
    if fizzbuzz % 3 == 0 and fizzbuzz % 5 == 0:
        print("fizzbuzz")
        continue
    elif fizzbuzz % 3 == 0:
        print("fizz")
        continue
    elif fizzbuzz % 5 == 0:
        print("buzz")
        continue
    print(fizzbuzz)

fizzbuzz
1
2
fizz
4
buzz
fizz
7
8
fizz
buzz
11
fizz
13
14
fizzbuzz
16
17
fizz
19
buzz
fizz
22
23
fizz
buzz
26
fizz
28
29
fizzbuzz
31
32
fizz
34
buzz
fizz
37
38
fizz
buzz
41
fizz
43
44
fizzbuzz
46
47
fizz
49
buzz


input:orange  →   output : puboif


In [3]:
def replaceCharacterType(str): 
    vowel = False
    consonant = False
    for i in range(0, len(str)):           
        ch = str[i]    
        if ( (ch >= 'a' and ch <= 'z') or 
             (ch >= 'A' and ch <= 'Z') ):  
            if(isVowel(ch)):
                vowel = True
            else:
                consonant = True
            if (vowel == True): 
                print(chr(ord(ch) + 1), end =" ")
                vowel = False
            else: 
                i=1
                while True:
                    tmp = chr(ord(ch)+i)
                    if (isVowel(tmp)):
                        print(tmp, end =" ")
                        break
                    else:
                        i +=1

def isVowel(ch):
    ch = ch.lower()   
    if (ch == 'a' or ch == 'e' or ch == 'i' 
            or ch == 'o' or ch == 'u'):
                return True
    else:
        return False
str = "orange"
replaceCharacterType(str)


p u b o i f 

Accept a string from the user and count no.of vowels,consonants and special characters

In [4]:
string = input("Please Enter Your Own String : ")
string = string.lower()
vowels = consonants = digits = special_chars = 0
for i in string:
    if( i>='a' and i<='z' ):
        if( i=='a' or i=='e' or i=='i'or i=='o'or i=='u' ):
            vowels += 1
        else:
                consonants += 1
    elif( i.isdigit() ):
            digits += 1
    else:
            special_chars += 1
print( "Vowels: ", vowels)    
print( "Consonants: ", consonants)
print( "Special Characters: ", special_chars)


Please Enter Your Own String : abc@95*
Vowels:  1
Consonants:  2
Special Characters:  2


 Test case1:
        string →  "((((()()()))))()"
        output →  8
    Test case2:
        string →  "((()))()"
        Output →  4


In [6]:
str=input("enter the string")
c1=0
c2=0
for i in range(0,len(str)):
  if(str[i]=='('):
    c1=c1+1
  else:
    c2=c2+1
if(c1==c2):
  print(c1)

enter the string((((()()()))))()
8


Accept input as a string and display sum of digits as the output 
input:Appli123cation456 
Output: 21


In [7]:
s=0
a=input("enter")
for i in a:
  if i.isdigit():
    s=s+int(i)
print(s)

enterak442
10


count the no.of pairs of a in the given string(Infosys InfyTQ)
    string →  "abbaaccbbaaa"
    output →  2


In [9]:
str=input("enter string")
count=str.count("ab")
print(count)

enter stringab
1


 Accept a number as an input and check whether the given number is palindrome or not
if it is a palindrome number print the number on the screen
if it is not a palindrome number reverse that number and add it to previous number repeat this until will get a palindrome number and print that palindrome number on the screen
input: 127
output: 848


In [12]:
r=0
i=int(input())
n=i
while(i>0):
  o=i%10
  r=r*10+o
  i=i//10
if(n==r):
  print(n)
else:
  print(n+r)

127
848
